### PR TITLE
PodSecurity: add namespace update verify benchmark

### LINKED ISSUE #103303: [PodSecurity] Scale testing of namespace bulk check behavior
- [ ] Add tests to verify timeouts & pod limits on namespace checks

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Fixes https://github.com/kubernetes/kubernetes/issues/103303

Adds a benchmark for updating a namespace containing 3000 pods that requires evaluation of the restricted pod security standard on all pods

```
BenchmarkVerifyNamespace-12    	      78	  13184868 ns/op	 5401213 B/op	   54050 allocs/op
```

Shows an informer-fed pod-lister taking ~.01 seconds to validate a namespace containing 3000 pods with the restricted policy.

```release-note
NONE
```

/cc @tallclair 
/sig auth

### CODE REVIEW

File: plugin/pkg/admission/security/podsecurity/testdata/pod_baseline.yaml [hunk 1]
```diff
@@ -0,0 +1,670 @@
+# this pod fixture is used for benchmarks and should be kept updated to pass the latest baseline policy
```
Comment (tallclair) [inline plugin/pkg/admission/security/podsecurity/testdata/pod_baseline.yaml:1]: Rename `pod.yaml` to `pod_restricted.yaml`?

**Review by tallclair — COMMENTED** [review]
/lgtm
/approve

File: plugin/pkg/admission/security/podsecurity/testdata/pod_baseline.yaml (hunk 1 above)
Comment (liggitt) [reply plugin/pkg/admission/security/podsecurity/testdata/pod_baseline.yaml:1]: done

**Pull request merged**